axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetFailOverControlServiceImpl.java

```diff
@@ -38,32 +49,37 @@ public class FixedAssetFailOverControlServiceImpl implements FixedAssetFailOverC
   public void controlFailOver(FixedAsset fixedAsset) throws AxelorException {
     Objects.requireNonNull(fixedAsset);
     FixedAssetCategory fixedAssetCategory = fixedAsset.getFixedAssetCategory();
+    BigDecimal grossValue =
+        currencyScaleServiceAccount.getCompanyScaledValue(fixedAsset, fixedAsset.getGrossValue());
+    BigDecimal importAlreadyDepreciatedAmount =
+        currencyScaleServiceAccount.getCompanyScaledValue(
+            fixedAsset, fixedAsset.getImportAlreadyDepreciatedAmount());
+    BigDecimal importFiscalAlreadyDepreciationAmount =
+        currencyScaleServiceAccount.getCompanyScaledValue(
+            fixedAsset, fixedAsset.getImportFiscalAlreadyDepreciatedAmount());
+    BigDecimal importIfrsAlreadyDepreciatedAmount =
+        currencyScaleServiceAccount.getCompanyScaledValue(
+            fixedAsset, fixedAsset.getImportIfrsAlreadyDepreciatedAmount());
+
     if (isFailOver(fixedAsset) && fixedAssetCategory != null) {
-      if (fixedAsset.getImportAlreadyDepreciatedAmount() != null
-          && fixedAsset.getImportAlreadyDepreciatedAmount().compareTo(fixedAsset.getGrossValue())
-              > 0) {
+      if (importAlreadyDepreciatedAmount != null
+          && importAlreadyDepreciatedAmount.compareTo(grossValue) > 0) {
```

Comment:
Maybe you can create a method to scale two values then return their comparison. I also saw some comparison in `FixedAssetImportServiceImpl` so it could be used there too. That would prevent all those scaled local variable from being created.

Reply:
Resolved